tools/rust/smessage_encryption.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright 2019 (c) rust-themis developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::{self, Read};
+use std::process::exit;
+
+use clap::clap_app;
+use themis::keys::{KeyPair, PublicKey, SecretKey};
+use themis::secure_message::{SecureMessage, SecureSign, SecureVerify};
+
+fn main() {
+    let matches = clap_app!(smessage_encryption =>
+        (version: env!("CARGO_PKG_VERSION"))
+        (about: "Secure Message multitool.")
+        (@arg command:    +required "Action: enc | dec | sign | verify")
+        (@arg secret_key: +required "Path to file with recipient's secret key")
+        (@arg public_key: +required "Path to file with sender's public key")
+        (@arg message:    +required "The message to process")
+    )
+    .get_matches();
+
+    let command = matches.value_of("command").unwrap();
+    let secret_key_path = matches.value_of("secret_key").unwrap();
+    let public_key_path = matches.value_of("public_key").unwrap();
+    let message = matches.value_of("message").unwrap();
+
+    let secret_key = read_file(&secret_key_path).expect("read secret key");
+    let secret_key = SecretKey::try_from_slice(secret_key).expect("parse secret key");
+    let public_key = read_file(&public_key_path).expect("read public key");
+    let public_key = PublicKey::try_from_slice(public_key).expect("parse public key");
+
+    match command {
+        "enc" => {
+            let key_pair = KeyPair::try_join(secret_key, public_key).expect("matching keys");
+            let encrypter = SecureMessage::new(key_pair);
+
+            // TODO: this reads strange, maybe we should use a better name (like, "encrypt")
+            let encrypted = encrypter.wrap(&message).unwrap_or_else(|error| {
```

Comment:
The reason of this name is that original API of SecureMessage is used either encrypt/decrypt or sign/verify the message.
https://github.com/cossacklabs/themis/wiki/Secure-Message-cryptosystem

As soon as in Rust you used SecureMessage, SecureSign and SecureVefiry as separate entities, it makes sense to rename wrap/unwrap into encrypt/decrypt